gui/velociraptor/src/components/forms/form.js

```diff
@@ -72,6 +75,95 @@ export default class VeloForm extends React.Component {
 
         // A Date() object that is parsed from value in local time.
         timestamp: null,
+        multichoices: undefined,
+    }
+
+    componentDidMount = () => {
+        this.source = axios.CancelToken.source();
+        this.fetchArtifacts(this.props.param.artifact_type);
+    }
+
+    componentWillUnmount() {
+        this.source.cancel();
+    }
+
+    fetchArtifacts(artifact_type) {
+        // Cancel any in flight calls.
+        this.source.cancel();
+        this.source = axios.CancelToken.source();
+
+        // Load all artifacts, but only keep the ones that match the
+        // specified type
+        api.post("v1/GetArtifacts",
+                {
+                    search_term: "...",
+                    // This might be too many to fetch at once but we
+                    // are still fast enough for now.
+                    fields: {
+                        name: true,
+                        type: true,
```

Comment:
Maybe we can set the type in here and have the server pre-filter these? this will reduce the total number sent.